src/main/java/com/epam/cdp/m2/hw2/aggregator/Java8Aggregator.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.cdp.m2.hw2.aggregator;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javafx.util.Pair;
+
+public class Java8Aggregator implements Aggregator {
+
+    @Override
+    public int sum(List<Integer> numbers) {
+        return numbers.stream()
+                .mapToInt(Integer::new)
+                .sum();
+    }
+
+    @Override
+    public List<Pair<String, Long>> getMostFrequentWords(List<String> words, long limit) {
+        List<Pair<String, Long>> result = words.stream()
+                .map(String::toLowerCase)
+                .collect(Collectors.groupingBy(w -> w,
+                        Collectors.summingLong(w -> 1L)))
+                .entrySet().stream()
+                .map(es -> new Pair<>(es.getKey(), es.getValue()))
+                .sorted(Comparator.comparing(Pair::getValue)) // TODO: Figure out why `Comparator.comparing(Pair::getValue).reversed()` throws error
+//                 .sorted(Comparator.reverseOrder())
+//                .limit(limit)
+                .collect(Collectors.toList());
```

Comment:
Let's try this:

        List<Pair<String, Long>> result = words.stream()
                .map(String::toLowerCase)
                .collect(Collectors.groupingBy(w -> w,
                        Collectors.summingLong(w -> 1L)))
                .entrySet().stream()
                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
                .map(es -> new Pair<>(es.getKey(), es.getValue()))
                .limit(limit)
                .collect(Collectors.toList());

Reply:
oh, nice. I didn't figured that we can actually sort it before even mapping to the list of pairs. very nice trick.
thanks a lot!